### DISCUSSION TITLE
Keep HLS available after source has been closed?

### DISCUSSION THREAD

**Discussion by Ernovation** [bluenviron/mediamtx#2500] [Questions]
### Question

My goal is to stream live video to MediaMTX using RTSP or RTMP and serve it through HLS. I would like to keep the HLS available after the live video has ended. Now, when close the source (e.g. ffmpeg), the HLS muxer is destroyed, index.m3u8 and stream.m3u8 are no longer available and all stored .mp4 files are removed (when I specify hlsDirectory in mediamtx.yml, which I do). I would like to keep all those files, including the .m3u8 files, available, preferably even after mediamtx stops, so that I can still serve the video to viewers who missed the live event. 
I tried a script to run at runOnNotReady, but when I try to get the index.m3u8 (using wget) it is no longer there. I think this happens because mediamtx does not wait for the runOnNotReady command to exit, and goes ahead closing cleaning up the hls muxer related to the path.
Do you have any suggestions how I could solve this problem? Am I missing something, or should I make some changes in the source to make this possible (e.g. a setting for HLS not destroy the muxer when the source is closed)?

**Comment by aler9** [comment]
Hello, this is a feature that is currently not present into the server, since the time dimension (i.e. recordings) has been introduced just some months ago. I've created a feature request here: #2716
